src/js/libcs/Render2D.js

```diff
@@ -373,6 +373,32 @@ Render2D.arrowShapes = {
     },
 };
 
+Render2D.onSegment = function(startpos, endpos, pt) {
+    var det = List.det(List.realMatrix([
+        [startpos.x, startpos.y, 1],
+        [endpos.x, endpos.y, 1],
+        [pt.x, pt.y, 1]
+    ]));
+    var onLine = CSNumber._helper.isAlmostEqual(det, CSNumber.zero, CSNumber.epsbig);
```

Comment:
Do we need this check? The way I see it, all use cases of this function will ensure collinearity of points, so the relative positions are the only important thing.